modules/drivers/bigquery-cloud-sdk/test/metabase/driver/bigquery_cloud_sdk/query_processor_test.clj

```diff
@@ -854,3 +855,37 @@
            (let [t         [:absolute-datetime #t "2022-04-22T18:27:00-08:00"]
                  hsql-form (sql.qp/add-interval-honeysql-form :bigquery-cloud-sdk t 3 :month)]
              (sql.qp/format-honeysql :bigquery-cloud-sdk hsql-form))))))
+
+(tx/defdataset text-coerced-to-date
+  [["coerced_dates"
+    [{:field-name "field_date_time", :base-type :type/Text, :coercion-strategy :Coercion/ISO8601->DateTime}
+     {:field-name "field_date",      :base-type :type/Text, :coercion-strategy :Coercion/ISO8601->Date}
+     {:field-name "field_time",      :base-type :type/Text, :coercion-strategy :Coercion/ISO8601->Time}]
+    [["2008-05-26T03:22:17" "2008-05-26" "03:22:17.357"]]]])
```

Comment:
Is there an existing dataset I could use instead of defining my own? Is there a better way of defining this dataset? (Most probably it won't be used in any other test.)

Reply:
that looks fine. I've done similar before in https://github.com/metabase/metabase/blob/master/test/metabase/query_processor_test/alternative_date_test.clj#L157 . You could try to combine those all into one dataset with multiple tables. Another alternative is to promote those all to a single edn file à la https://github.com/metabase/metabase/tree/master/test/metabase/test/data/dataset_definitions

Reply:
I'll have a look if I can just add BigQuery to those tests of yours.